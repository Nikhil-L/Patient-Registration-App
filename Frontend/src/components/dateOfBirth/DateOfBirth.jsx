import React, { Component } from 'react';
import DatePicker from 'react-datepicker';
import "react-datepicker/dist/react-datepicker.css";

/**
 * This component is used to create a input option to enter date of birth
 */
export class DateOfBirth extends Component {

  /**
   * Initializes the props
   * @param {*} props 
   */
  constructor (props) {
    super(props)
    this.state = {
      startDate: new Date()
    };
    this.props.handleChange(this.state.startDate);
    this.handleChange = this.handleChange.bind(this);
  }

  /**
   * This method is used to handle the changes made on the browser
   * @param {*} date 
   */
  handleChange(date) {
    this.setState({
      startDate: date
    });
    this.props.handleChange(date);
  }

 /**
  * This is used to render the component to the webpage
  * @returns a html code 
  */
  render() {
    return (
      <div className="form-group">
        <div>
          <label name = "DateOfBirth">Date of Birth</label>
        </div>
        <DatePicker
            selected={ this.state.startDate }
            onChange={ this.handleChange }
            name="startDate"
            maxDate = {new Date()}
            dateFormat="MM/dd/yyyy"
        />
      </div>

    );
  }
  
}
