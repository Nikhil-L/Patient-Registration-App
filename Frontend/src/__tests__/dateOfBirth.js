import React from "react";
import {DateOfBirth} from "../components/dateOfBirth/DateOfBirth"
import renderer from "react-test-renderer"
import {shallow} from "enzyme"

/**
 * This is to test the dateOfBirth Component
 */
describe("DateOfBirth test", ()=>{
  
  /**
   * Snapshot testing
   */
  it("matches a snapshot", ()=>{
    const tree = renderer.create(<DateOfBirth handleChange={jest.fn()}/>).toJSON();
    expect(tree).toMatchSnapshot();
  })

  /**
   * event testing
   */
  it('event testing', ()=>{
    let wrapper = shallow(<DateOfBirth handleChange={jest.fn()}/>);
    wrapper.find({name:'startDate'}).simulate('change', {target:
      {name:'startDate', value:'07/03/2000'}})
  })
})