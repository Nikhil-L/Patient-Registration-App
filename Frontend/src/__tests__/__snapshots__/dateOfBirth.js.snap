// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DateOfBirth test matches a snapshot 1`] = `
<div
  className="form-group"
>
  <div>
    <label
      name="DateOfBirth"
    >
      Date of Birth
    </label>
  </div>
  <div
    className="react-datepicker-wrapper"
  >
    <div
      className="react-datepicker__input-container"
    >
      <input
        className=""
        disabled={false}
        name="startDate"
        onBlur={[Function]}
        onChange={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        readOnly={false}
        type="text"
        value="07/07/2021"
      />
    </div>
  </div>
</div>
`;
